test/common/http/conn_manager_impl_test.cc

```diff
@@ -1194,6 +1194,27 @@ TEST_F(HttpConnectionManagerImplTest, RouteShouldUseNormalizedHost) {
   filter_callbacks_.connection_.raiseEvent(Network::ConnectionEvent::RemoteClose);
 }
 
+// Observe host header w/o trailing dot in host when trailing dot removal is configured.
+TEST_F(HttpConnectionManagerImplTest, StripTrailingHostDot) {
```

Comment:
Mind adding a test where we set this config option but pass through a header without a trailing dot?

Reply:
Sure. will add.